backend/src/rate_limiter/provider.rs

```diff
@@ -80,13 +74,16 @@ pub(crate) mod persistence {
             connection: impl PgExecutor<'_>,
             key: &RlKey,
         ) -> Result<(), StorageError> {
+            // Ensure valid_until is rounded to microseconds, since postgres
+            // only supports microsecond precision.
+            let valid_until = self.valid_until.round_subsecs(6);
```

Comment:
Another possibility would be to make sure that `Allowance` always contains truncated `valid_until`. Otherwise, PartialEq impl for it might behave funny.

Reply:
Yes, that's what I'm implementing right now.